src/NodaTime/DateInterval.cs

```diff
@@ -134,6 +134,24 @@ public bool Contains(LocalDate date)
             return Start <= date && date <= End;
         }
 
+        /// <summary>
+        /// Checks whether the given interval is within this interval. This requires that the start date of the specified
+        /// interval is not earlier than the start date of this interval, and the end date of the specified interval is not
+        /// later than the end date of this interval.
+        /// </summary>
+        /// <remarks>
+        /// An interval contains another interval with same start and end dates, or itself.
+        /// </remarks>
+        /// <param name="interval">The interval to check for containment within this interval.</param>
+        /// <exception cref="ArgumentException">Start and end dates of <paramref name="interval"/> are not in the same
+        /// calendar as the start and end date of this interval.</exception>
+        /// <returns><c>true</c> if <paramref name="interval"/> is within this interval; <c>false</c> otherwise.</returns>
+        public bool Contains([NotNull] DateInterval interval)
+        {
+            ValidateInterval(interval);
```

Comment:
Possibly optimize for `interval == this`? Then you don't need to do it in Intersection